std/experimental/normint.d

```diff
@@ -0,0 +1,571 @@
+    }
+    unittest
+    {
+        // up cast
+        static assert(cast(NormalizedInt!ushort)NormalizedInt!ubyte(0xFF) == 0xFFFF);
+        static assert(cast(NormalizedInt!ushort)NormalizedInt!ubyte(0x81) == 0x8181);
+
+        // down cast
+        static assert(cast(NormalizedInt!ubyte)NormalizedInt!ushort(0xFFFF) == 0xFF);
+        static assert(cast(NormalizedInt!ubyte)NormalizedInt!ushort(0x9F37) == 0x9F);
+
+        // signed -> unsigned
+        static assert(cast(NormalizedInt!ubyte)NormalizedInt!byte(127) == 0xFF);
+        static assert(cast(NormalizedInt!ushort)NormalizedInt!byte(127) == 0xFFFF);
+        static assert(cast(NormalizedInt!ubyte)NormalizedInt!byte(-127) == 0);
+        static assert(cast(NormalizedInt!ubyte)NormalizedInt!byte(-128) == 0);
+        static assert(cast(NormalizedInt!ushort)NormalizedInt!byte(-127) == 0);
+        static assert(cast(NormalizedInt!ubyte)NormalizedInt!short(-32767) == 0);
+        static assert(cast(NormalizedInt!ubyte)NormalizedInt!short(-32768) == 0);
+
+        // unsigned -> signed
+        static assert(cast(NormalizedInt!byte)NormalizedInt!ubyte(0xFF) == 0x7F);
+        static assert(cast(NormalizedInt!byte)NormalizedInt!ubyte(0x83) == 0x41);
+        static assert(cast(NormalizedInt!short)NormalizedInt!ubyte(0xFF) == 0x7FFF);
+        static assert(cast(NormalizedInt!short)NormalizedInt!ubyte(0x83) == 0x41C1);
+        static assert(cast(NormalizedInt!byte)NormalizedInt!ushort(0xFFFF) == 0x7F);
+        static assert(cast(NormalizedInt!byte)NormalizedInt!ushort(0x83F7) == 0x41);
+
+        // signed -> signed
+        static assert(cast(NormalizedInt!short)NormalizedInt!byte(127) == 32767);
+        static assert(cast(NormalizedInt!byte)NormalizedInt!short(32767) == 127);
+        static assert(cast(NormalizedInt!short)NormalizedInt!byte(-127) == -32767);
+        static assert(cast(NormalizedInt!byte)NormalizedInt!short(-32767) == -127);
+    }
+
+    F opCast(F)() const if(isFloatingPoint!F)
+    {
+        F r = value * F(1.0/max);
+        static if(isSigned!I)
+        {
+            // max(c, -1) is the signed conversion followed by D3D, OpenGL, etc.
+            r = std.algorithm.max(r, F(-1.0));
+        }
+        return r;
+    }
+    unittest
+    {
+        static assert(cast(float)NormalizedInt!ubyte(0xFF) == 1.0);
+        static assert(cast(float)NormalizedInt!ubyte(0x00) == 0.0);
+        static assert(cast(float)NormalizedInt!ubyte(0x80) > 0.5);
+        static assert(cast(float)NormalizedInt!ubyte(0x7F) < 0.5);
+        static assert(cast(float)NormalizedInt!byte(127) == 1.0);
+        static assert(cast(float)NormalizedInt!byte(-128) == -1.0);
+        static assert(cast(float)NormalizedInt!byte(-127) == -1.0);
+        static assert(cast(float)NormalizedInt!byte(0x00) == 0.0);
+        static assert(cast(float)NormalizedInt!byte(0x40) > 0.5);
+        static assert(cast(float)NormalizedInt!byte(0x3F) < 0.5);
+    }
+}
+
```

Comment:
The operator overloads need some kind of documentation. IMO adding a example to the `NormalizedInt` struct showing all supported operators would be fine.